### DISCUSSION TITLE
중급 프로젝트 2주차 팀회고

### DISCUSSION THREAD

**Discussion by sulyoungoh** [codeit-bootcamp-frontend/3-Weekly-Mission#338] [회고]
중급 프로젝트 2주차 팀 회고를 하고([회고 방식 상세보기](https://github.com/codeit-bootcamp-frontend/3-Weekly-Mission/discussions/337#discussion-6334449)), 이전처럼 댓글 남겨주세요🙏

아래 내용들을 토대로 이번 팀 회고에서 점검해보면 좋을 것 같아요. 
- 프로젝트 2주차이지만 여전히 정리 되지 않은 팀 규칙, 컨벤션이 있다면 이번 시간에 정리해보는 것도 좋을 것 같아요. 남은 프로젝트 기간에 더 좋은 협업을 할 수 있을거예요. 
- 팀 소통 방식 
- 협업 관계에서 느낀 감정 
- 기타 등등

**Comment by yulmai999** [comment]
# 5팀

# Keep

- 꼼꼼하게 코드리뷰를 해주고 세세하게 신경써서 코드를 짜는점
- 더 나은 코드를 작성하기 위해 꾸준히 의견을 주고 받는점이 좋았습니다.
- PR을 꼼꼼하게 리뷰하고 프로젝트 구조에 대해 생각하며 진행하는 점이 좋았습니다.
- 새로운 기술의 사용법과 세팅, 에러사항을 공유하고 협업하는과정이 좋았습니다.

## Problem

- 프로젝트에 새로운 기술들을 많이 쓰게되어서 적응에 시간과 노력이 필요함
- 에러가 나는 부분에 너무 시간을 빼앗김
- 한 커밋과 pr 내에 코드 양이 생각보다 많이졌음.

## Try

- 거부감없이 새로운기술을 익히고 써보기, 꼼꼼하게 일하기
- 에러가 해결이 너무 안돼면 팀원에게 도움 요청하기
- 커밋과 pr이 너무 비대해지지 않도록 미리 순서를 정해 코드를 작성/수정하도록 해보기
- 협업에 필요한사항은 빨리빨리공유하기, 베이스코드를 바꿀때는 팀원과 공유하기

**Comment by GihoKo** [comment]
# **👍🏻**Keep

- 모르는 부분에 대해 물어보면 설명을 잘 해주셔서 좋습니다.(예시를 들어서 설명해주셔서 더 이해가 잘 됩니다.)
- PR에 자세한 리뷰가 달려서 코드의 품질 향상에 도움이 됩니다.
- 서로 깊은 고민을 공유해서 많은 것을 깨닫습니다.
- 설명이 길어지면 중간에 놓칠 때가 있는데, 이해했는지 체크해주시면서 설명해주셔서 좋았습니다.

## **🤔**Problem

- 결정을 내리기 어려워 하는 것 같습니다.
- 회의를 할 때 팀원이 어디까지 이해했는지 정확하게 판단하기가 힘듭니다.
- PR이 단위를 크게 해서 작업이 얼마나 진행이 되는지 알기가 어렵습니다.
- 시도해보려는게 많아서 무언가를 바꾸려고 할 때의 비용이 크게 느껴집니다.

## **🔥**Try

- 팀미팅 시작 전 서로 어디까지 진행했는지 이야기 해보는 것도 좋을 것 같습니다.
- PR 리뷰가 늦어지다보니 작업 속도에도 영향이 있는 것 같습니다. 다음부터는 PR 리뷰 페어를 구성해서 어느정도의 강제성이 필요해 보입니다.
- 작업 단위를 나누어서 빠르게 PR을 올리는 것일 좋을 듯 합니다.
- 미팅 시 이야기할거리를 숙지하고 미팅이 너무 루즈해지지 않게 의견을 내는 걸 노력해야겠습니다.

**Comment by 2zzzyoung** [comment]
# 4팀🐼
## Keep
- 소통이 원활하고 참석이 잘 된다.
- 작업 템포가 적절하다.
- 부족한 인원을 잘 끌어준다.
- 팀원 분들이 많은 도움을 주신다.

## Problem
- 배포할 때 오류 처리를 빨리 해결해야 할 것 같다.
- PR 시 코드 리뷰가 있었으면 좋겠다.
- 개발할 때 더 많은 의사소통이 필요할 것 같다.
- 내가 맡은 기능 빨리 끝내기.

## Try
- 완성 후에 파일, 폴더 이름을 정리하면 좋을 것 같다.
- 상호 협의 하에 피그마에 없던 추가적인 기능을 구현하면 좋을 것 같다.
- 다른 분들의 코드를 보고 많이 배워야겠다.

**Comment by mingzzi96** [comment]
# 7팀
## KEEP
- 열정적이고 도움을 주려고 노력한다
- 빠른 PR 승인
- 매번 화이팅으로 끝내는거 너무 좋다.
- 내 파트가 아니더라도 다들 자기일처럼 봐주려고 하는 자세가 좋았다
- 매일 진행상황을 제대로 체크하고 배분이 확실하게 되어 좋다. 

## PROBLEM
- 조금이라도 궁금하거나 이상한 것이 있으면 바로바로 공유되면 좋겠다.
- 맡은 페이지 진도가 느린게 조금아쉬웠다.
- 간단한 작업이라고 생각한것도 막상 시작하면 예상보다 시간이 오래걸렸다.

## TRY
- 우선 주어진 기능 구현을 목표로하고 리펙토링해 보면 좋겠다.
- 조금 속도를 더 내보자.
- 한번 구현해봤던 기능은 다음에 구현할때 더 빠르게 해낼수 있도록 노력하기

**Comment by soyxxn** [comment]
🤓 3팀
=============
중급 프로젝트 2주차 만족도 ⭐2.25 / 5
-------------

### KEEP
-  다들 열심히 하는 모습에 자극이 되었다.
- 코드 리뷰를 열심히 주고 받으면서 배우는 것도 있고 오류가 생길 수 있는 부분도 미리 잡을 수 있어 좋았다.
- 코어타임이 잘 지켜져서 좋다.
- 수면시간을 잘 지키려고 노력했다.

### PROBLEM
- PR 승인이 좀 느린 것 같다 디스코드 알림이 뜨면 바로 가서 하는 것도 좋을 것 같다.
- 한 가지 일에 몰입하느라 작업 속도가 늦어진 것 같다.
- 연관성이 있는 기능도 꽤 있기 때문에 기능 구현의 우선순위를 잘 정하고 지켜야 할 것 같다.
- 어떤 이슈나 궁금증이 생겼을 때 너무 혼자 가지고 있는 것 같아 공유를 해봐야겠다.
- 뭔가 애매한데 어떻게 해야 할 지 고민이 되어 물어봐야지 했던 지점을 일단 해결하고 까먹는 경우가 많다.

### TRY
- 프로젝트를 하면서 궁금하거나 어려웠던 점을 문서화 해보기
- 체력이 굉장히 떨어진 것을 느껴 하루 30분 산책이라도 하기
- 일단 있는 기능들은 우선순위를 정하고 빨리 완성을 한 후 부족한 부분 고치기
- 코드리뷰 열심히 하기
- 기능 개발 중이면 기능 개발만 하고 도중에 다른 리팩토링 하지 않기

**Comment by DanteSnow** [comment]
# 🔥 2팀 🔥

# Keep
- PR 요청시 서로 코드리뷰 해주며 팀원이 어떻게 작업했는지 확인 할 수 있는 것.
- github의 기능으로 프로젝트의 전반적인 진행상황을 확인하는 것.
- 각자의 진행상황에 맞게 스케줄을 잘 관리했고, 폴더 구조에 맞게 작업을 진행하는 것이 좋았습니다. 또, 서로 잘 모르는 부분에 있어서 항상 자세히 팀원들끼리 알려주고, 어려운 부분은 같이 해결하려는 모습이 좋았습니다.
- 각자 맡은 역할에 대해서 최선을 다해 임하는 모습이 좋았습니다.
- 다른 팀원의 문제도 자신의 문제처럼 같이 들여다 보는 모습이 좋았습니다.


# Problem
- 프로젝트의 효율적인 진행을 위해서 컨벤션 또는 쓰기로 한 기능을 제외하고 나중에 리팩토링으로 미루어 놓는 것
- 개인적으로 tailwind 사용법에 많이 익숙하지 않아, 스타일링 작업을 뒤로 미루었는데, 여러 스타일링을 한번에 리팩토링해야 하다보니 더 시간도 많이 걸리는 것 같습니다.
- 기능을 구현하기 위한 선행 학습이 부족해서 정석적인 방법이 아닌 돌아가는 방법을 택했습니다. (Next14, 쿠키, 인증 등)


# Try
- 익숙치 않은 기능과 컨벤션이라도 최대한 맞춰가려고 하며 프로젝트를 진행해 본다
- 기본적인 전체 스타일을 우선적으로 적용시키고, 어려운 부분은 주석을 남겨놓고, 나중에 수정해야 하는 부분을 파악해 놓도록 하겠습니다.
- 이번 기회에 확실하게 학습한다고 생각하며 최선을 다해 임할 것입니다.

**Comment by jihee1103** [comment]
## 1팀

# Keep

- 서로 겹치는 부분에 있어서 소통이 원활함
- 프로젝트 진행이 더디지 않도록 다들 노력하는 모습이 보임
- 모르겠다 싶은 부분 공유가 잘 되는 듯함

# Problem

- 폴더 정리가 잘 되지 않는거 같음. git push와 git pull을 통한 폴더 정리가 부족한 거 같음.
- pr 올릴 때 조금 더 확인하고 올리면 좋을 것 같음(내 자신에게도 하는 말)
- 코드짜기에도 급급해 pr리뷰를 잘 하지 못한다는 것
- 폴더관리가 제대로 되고있지 않는 것 같다.   필요한 요소들을 고려해서 모든 폴더를 만들어놓고 어디에 무엇을 넣을지 고민을 다같이 했으면 좋았을 것 같다.


# Try

- 소통을 통해서 필요없는 부분과 변경사항에 대해서 확실하게 정해서 폴더가 어지러워지지 않도록 노력해야 될 거 같음.
- 조금 더 꼼꼼하게 코드를 짜고 확인했다고 생각해도 한 번 더 확인해보기
